sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -859,11 +859,11 @@ case class FromUnixTime(sec: Expression, format: Expression, timeZoneId: Option[
     } else {
       val zid = ctx.addReferenceObj("zoneId", zoneId, classOf[ZoneId].getName)
       val locale = ctx.addReferenceObj("locale", Locale.US)
-      val tf = TimestampFormatter.getClass.getName.stripSuffix("$")
+      val tf = classOf[Iso8601TimestampFormatter].getName
```

Comment:
Should we add a comment (maybe TODO) with mentioned bug report. So, we could revert this back when the issue will be fixed?

Reply:
From my understanding, it's not a bug in the JVM; the failure is (now) correct, so we need to fix here.

Reply:
Ya. This is a JVM spec.

Reply:
Ya. This is a JVM spec.

Reply:
Ya. This is a JVM spec.